public/app/features/variables/state/actions.test.ts

```diff
@@ -267,4 +281,161 @@ describe('shared actions', () => {
+      const startRefreshMock = jest.fn();
+      const adapter = createIntervalVariableAdapter();
+      adapter.updateOptions = args.throw
+        ? jest.fn().mockRejectedValue('Something broke')
+        : jest.fn().mockResolvedValue({});
+      variableAdapters.set('interval', adapter);
+      variableAdapters.set('constant', createConstantVariableAdapter());
+
+      // initial variable state
+      const initialVariable = variableMockBuilder('interval')
+        .withUuid('0')
+        .withName('interval-0')
+        .withOptions('1m', '10m', '30m', '1h', '6h', '12h', '1d', '7d', '14d', '30d')
+        .withCurrent('1m')
+        .withRefresh(VariableRefresh.onTimeRangeChanged)
+        .create();
+
+      // the constant variable should be filtered out
+      const constant = variableMockBuilder('constant')
+        .withUuid('1')
+        .withName('constant-1')
+        .withOptions('a constant')
+        .withCurrent('a constant')
+        .create();
+      const initialState = {
+        templating: { variables: { '0': { ...initialVariable }, '1': { ...constant } } },
+        dashboard,
+      };
+
+      // updated variable state
+      const updatedVariable = variableMockBuilder('interval')
+        .withUuid('0')
+        .withName('interval-0')
+        .withOptions('1m')
+        .withCurrent('1m')
+        .withRefresh(VariableRefresh.onTimeRangeChanged)
+        .create();
+
+      const variable = args.update ? { ...updatedVariable } : { ...initialVariable };
+      const state = { templating: { variables: { '0': variable, '1': { ...constant } } }, dashboard };
+      const getStateMock = jest
+        .fn()
+        .mockReturnValueOnce(initialState)
+        .mockReturnValue(state);
+      const dispatchMock = jest.fn();
+
+      return {
+        range,
+        dependencies,
+        dispatchMock,
+        getStateMock,
+        updateTimeRangeMock,
+        templateVariableValueUpdatedMock,
+        startRefreshMock,
+        emitMock,
+      };
+    };
+
+    describe('and options are changed by update', () => {
+      it('then correct dependencies are called', async () => {
+        const {
+          range,
+          dependencies,
+          dispatchMock,
+          getStateMock,
+          updateTimeRangeMock,
+          templateVariableValueUpdatedMock,
+          startRefreshMock,
+          emitMock,
+        } = getOnTimeRangeUpdatedContext({ update: true });
+
+        await onTimeRangeUpdated(range, dependencies)(dispatchMock, getStateMock, undefined);
+
+        expect(dispatchMock).toHaveBeenCalledTimes(0);
+        expect(getStateMock).toHaveBeenCalledTimes(4);
+        expect(updateTimeRangeMock).toHaveBeenCalledTimes(1);
+        expect(updateTimeRangeMock).toHaveBeenCalledWith(range);
+        expect(templateVariableValueUpdatedMock).toHaveBeenCalledTimes(1);
+        expect(startRefreshMock).toHaveBeenCalledTimes(1);
+        expect(emitMock).toHaveBeenCalledTimes(0);
+      });
```

Comment:
Not sure about the value of these tests, maybe these tests should be e2e tests instead.

Reply:
yea, if we have a testdata that can have variables that update based on time we could have an e2e test for it: 

Created this PR that adds a dashboard that could be used in such an e2e test:
https://github.com/grafana/grafana/pull/22829